src/angular-app/languageforge/lexicon/shared/fit-text.directive.ts

```diff
@@ -0,0 +1,33 @@
+import * as angular from 'angular';
+
+export class FitTextDirective implements angular.IDirective {
+
+  link(scope: angular.IScope, element: angular.IAugmentedJQuery, attr: angular.IAttributes) {
+    let kInput: boolean = false;
+    function updateHeight(): any {
+      let height = element.prop('scrollHeight');
+      element.height = height;
+      element.css({ 'max-height': height + 'px' });
+      element.css({ height: height + 'px' });
```

Comment:
I'm unsure what the performance cost of executing element.css on every keystroke.  I wonder if we should compare the existing height to the scroll-height first, and only if they are different set them via element.css().

```suggestion
      if (element.height != height) {
        element.height = height;
        element.css({ 'max-height': height + 'px' });
        element.css({ height: height + 'px' });
      }
```

Also, is element.css dependent on JQuery ?  It kind of looks that way.  If so, is there a way to not use any JQuery in this directive?